jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/api/client/clientrequestcontext/JAXRSClientIT.java

```diff
@@ -860,6 +861,53 @@ protected void checkFilterContext(ClientRequestContext context)
     assertContains(entity, "Accept");
   }
 
+  /*
+   * @testName: containsHeaderStringTest
+   * 
+   * @assertion_ids:  JAXRS:JAVADOC:1353; JAXRS:JAVADOC:1354; 
+   * 
+   * @test_Strategy: Check if the specified header contains a specified value.
+   * 
+   * ClientRequestFilter.abortWith
+   */
+  @Test
+  public void containsHeaderStringTest() throws Fault {
+    ContextProvider provider = new ContextProvider() {
+      @Override
+      protected void checkFilterContext(ClientRequestContext context) throws Fault {
+          assertTrue(context.containsHeaderString("header1", "value"::equalsIgnoreCase));
+          assertTrue(context.containsHeaderString("HEADER1", ",", "value2"::equals));
+          //Incorrect separator character
+          assertFalse(context.containsHeaderString("header1", ";", "value2"::equalsIgnoreCase));
+          //Shouldn't find first value when separator character is incorrect
+          assertFalse(context.containsHeaderString("header1", ";", "value1"::equalsIgnoreCase));
+          //Test regular expression
+          assertTrue(context.containsHeaderString("header1", ";|,", "value2"::equalsIgnoreCase));
+          //White space in value not trimmed
+          assertFalse(context.containsHeaderString("header1", "whitespace"::equalsIgnoreCase));
+          //Multiple character separator
+          assertTrue(context.containsHeaderString("header2", "::", "Value5"::equalsIgnoreCase));
+          //Test default separator is comma
+          assertFalse(context.containsHeaderString("header3","value6"::equalsIgnoreCase));
```

Comment:
I cannot see how this line actually tests what its comment says.

Reply:
This test is ensuring that if a separator parameter is not set, that it will default to comma and containsHeaderString in this case will return "false" because ";" is the separator.   However, it is somewhat redundant so I have no problem removing it.

Reply:
I am not convinced that this test actually proofs usage of tha specific default separator. IMHO it only proofs that this lines fails, independent of the cause.